android-n/CommitContentSampleIME/CommitContentSampleIME/ImageKeyboard.cs

```diff
@@ -0,0 +1,266 @@
+﻿/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Android.App;
+using Android.Content;
+using Android.InputMethodServices;
+using Android.OS;
+using Android.Support.V13.View.Inputmethod;
+using Android.Support.V4.Content;
+using Android.Util;
+using Android.Views;
+using Android.Views.InputMethods;
+using Android.Widget;
+using Java.IO;
+using Java.Lang;
+using File = Java.IO.File;
+
+namespace CommitContentSampleIME
+{
+	[Service(Permission = "android.permission.BIND_INPUT_METHOD")]
+	[IntentFilter(new [] { "android.view.InputMethod" })]
+	[MetaData("android.view.im", Resource = "@xml/method")]
+	public class ImageKeyboard : InputMethodService
+	{
+		private const string Tag = "ImageKeyboard";
+		private const string Authority = "com.example.android.supportv13.sampleime.inputcontent";
```

Comment:
The upstream sample does have a bug, however we can easily resolve this:

1) For consistency, change the package name of the project - `com.xamarin.commitcontentsampleime`.
2) Change line 39 to the following:
        `const string Authority = "com.xamarin.commitcontentsampleime.inputcontent";`
3) Include the following missing Provider child element to the Application element in AndroidManifest.xml:
```
    <provider android:name="android.support.v4.content.FileProvider" android:authorities="com.xamarin.commitcontentsampleime.inputcontent" android:exported="false" android:grantUriPermissions="true">
        <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/file_paths" />
    </provider>
```